design/FY2021/NFP-Spaces.md

```diff
@@ -0,0 +1,234 @@
+especially for space-based interfaces (such as OpenStudio) and for codes and standards modeling.
+
+There are also computing performance advantages to dividing zones into smaller enclosures for radiant exchange.
+
+## Overview ##
+
+### Current EnergyPlus model ###
+EnergyPlus input currently has the following (very flat) heierarchy:
+
+    Zone
+        Surfaces (references a zone name)
+        People, Lights, etc. (references a zone name or zone list name)
+        Thermostat (one per zone)
+        HVAC equipment (attaches to a zone)
+
+Zones represent an air mass (air node) that exchanges heat with surfaces, internal loads, HVAC equipment, etc. 
+A Zone is essentially the "atomic unit" of the building. By default a zone has a uniform air temperature, but there are room air models 
+that allow modeling of stratification and other non-uniform temperature effects. Thermostats and HVAC equipment are assigned per Zone.
+
+The internal data model adds another layer:
+
+    Enclosure
+        Zone
+            Surfaces
+            People, Lights, etc.
+            Thermostat
+            HVAC equipment
+        Zone
+            Surfaces
+            People, Lights, etc.
+            Thermostat
+            HVAC equipment
+
+Enclosures are used for radiant and solar/daylighting exchange. Enclosures are primarily concerned with Surfaces, but they
+are also related to internal gains (which cast radiant and visible energy into the enclosure). By default there is one Enclosure for each
+Zone.  By using Construction:AirBoundary, multiple zones may be grouped into a single larger Enclosure. 
+Enclosures are assigned automatically based on the zones connected by an air boundary surface. 
+
+
+## New Space Object
+
+### Relationship assumptions ##
+
+If we keep the current definition of "Zone" and add the concept of
+"Space" (<= Zone), then this would add a new layer to the data model:
+
+    Zone
+        Thermostat
+        HVAC equipment
+        Space 2
+            Surfaces
+            Internal Gains (People, Lights, etc.)
+        Space 3
+            Surfaces
+            Internal Gains (People, Lights, etc.)
+
+    Enclosure
+        Space 1
+            Surfaces
+            Internal Gains (People, Lights, etc.)
+        Space 2
+            Surfaces
+            Internal Gains (People, Lights, etc.)
+    
+
+### Definitions
+
+ * Surface - A geometric plane which is attached to a Space. 
+   * A Surface can be opaque, transparent, or an air boundary.
+   * Each Surface belongs to one Space.
+
+ * Space - A collection of one or more Surfaces and internal gains.
+   * Each Space belongs to one Zone (explicitly user-assigned).
+   * The Spaces in a Zone are lumped together for the Zone heat balance.
+   * There is no heat balance at the Space level.
+   * Each Space belongs to one Enclosure (implicitly assigned).
```

Comment:
If the Space is implicitly assigned to an Enclosure then won't Spaces just be one-to-one with Enclosures?